Issues when including `music21j` in my project

Hello, I'm trying to use `music21j` in my project.

I'm a bit new to the node environment, but I figured out that using `webpack` was convenient for my purpose, so I created my project and then I run:

    npm install --save music21j

Then in one of my source files I tried:

    import { note } from "music21j"

But when I build the project using `npx webpack` I get various resolve errors like the following:

```
ERROR in ./node_modules/music21j/src/music21_modules.js
Module not found: Error: Can't resolve './music21/articulations' in '/home/matteo/git/nuvola/node_modules/music21j/src'
 @ ./node_modules/music21j/src/music21_modules.js 48:0-57 91:0-141:2
 @ ./src/core/melody/melody.js
 @ ./src/main.js
```

So I resorted to use the already built version provided (which however seems a bit of a hack to me, the above example should just work), by using

    import { note } from "music21j/releases/music21.debug"

I suspect that the issue might be related to the fact that the modules are written in typescript, but typescript doesn't get added to my dependencies nor get configured in the proper way. I don't know which is the proper way to handle this thing.

---

The second problem I'm facing is related with MIDI soundfonts. If I just build like in the above example I don't get any playback, and in the browser console I see the following error:

    Could not load soundfont; path was ../soundfonts/midi-js-soundfonts-master/FluidR3_GM//acoustic_grand_piano-ogg.js

The hack I found to make it work locally is to copy the files from `./node_modules/music21j/soundfonts` to `./dist/soundfonts` (I automated it using `copy-webpack-plugin`), but it doesn't work when I deploy to GitHub Pages (because of the level up `..` at the beginning of `../soundf....` and because I'm deploying a project page, so my base url is `mttbernardini.github.io/myproject`).

Is there a proper way to change the path of the soundfonts? I didn't find anything in the documentation and it seems to be hardcoded somewhere.

---

Let me know if you need more details, I'm willing to help.

Thank you in advance.

```
ERROR in ./src/core/melody/melody.js
Module not found: Error: Can't resolve 'music21' in '/home/matteo/git/nuvola/src/core/melody'
 @ ./src/core/melody/melody.js 2:0-68 33:18-23 60:18-23 76:37-45 82:13-18 91:14-18 106:14-20 111:15-19
 @ ./src/main.js
```

I'm not sure about using `music21`, the `package.json` file clearly states that the name of the package is `music21j`. Am I missing something?

As for the second issue, I managed to fix it in the following way. Is it the way it should be done, though?

```
import { common } from "music21j/releases/music21.debug"
common.urls.soundfontUrl = "./soundfonts/midi-js-soundfonts-master/FluidR3_GM/"
```

Thank you again

--- a/.gitignore
+++ b/.gitignore
@@ -2,8 +2,7 @@
 node_modules/
 node_modules/grunt/package.json
 
-build/*.js
-build/*.map
+build/
 
 tests/music21.tests.js
 tests/music21.tests.js.map

--- a/.gitignore
+++ b/.gitignore
@@ -1,13 +1,18 @@
-#node
+# node
 node_modules/
 node_modules/grunt/package.json
 
+# Build is not committed, just releases/
 build/
 
+# Built tests
 tests/music21.tests.js
 tests/music21.tests.js.map
 
+# Python files
+__pycache__
 *.py[cod]
+
 .buildpath
 
 # emacs
@@ -17,9 +22,11 @@ tests/music21.tests.js.map
 # vim
 *.swp
 
-#Eclipse
+# Eclipse
 /.settings
 .settings
+
+# IDEA
 .project
 .pydevproject
 
@@ -42,7 +49,6 @@ develop-eggs
 .installed.cfg
 lib
 lib64
-__pycache__
 
 # Installer logs
 pip-log.txt
@@ -69,8 +75,3 @@ node_modules/grunt-contrib-uglify/package.json
 .idea/music21j.iml
 .idea/vcs.xml
 .idea/workspace.xml
-
-
-# Built tests
-tests/music21.tests.js
-tests/music21.tests.js.map

--- a/Gruntfile.js
+++ b/Gruntfile.js
@@ -28,9 +28,6 @@ module.exports = grunt => {
     // const TARGET_MIN = path.join(BUILD_DIR, 'music21.min.js');
 
     const SOURCES = [
-        'src/music21_modules.js',
-        'src/music21/*.js',
-        'src/music21/*/*.js',
         'src/music21/*.ts',
         'src/music21/*/*.ts',
     ];
@@ -171,9 +168,6 @@ module.exports = grunt => {
         jsdoc: {
             dist: {
                 src: [
-                    'src/music21_modules.js',
-                    'src/music21/*.js',
-                    'src/music21/*/*.js',
                     'src/music21/*.ts',
                     'src/music21/*/*.ts',
                     'README.md',

--- a/package.json
+++ b/package.json
@@ -2,7 +2,8 @@
   "name": "music21j",
   "version": "0.11.11",
   "description": "A toolkit for computer-aided musicology, Javascript version",
-  "main": "releases/main.js",
+  "main": "releases/music21.debug.js",
+  "typings": "releases/main.d.ts",
   "files": [
     ".editorconfig",
     ".eslintignore",

--- a/package.json
+++ b/package.json
@@ -15,8 +15,7 @@
     "/ext",
     "/releases",
     "/scripts",
-    "/src/music21",
-    "/src/*.js",
+    "/src",
     "/testHTML",
     "/tests",
     "/webResources"

--- a/package.json
+++ b/package.json
@@ -2,7 +2,7 @@
   "name": "music21j",
   "version": "0.11.11",
   "description": "A toolkit for computer-aided musicology, Javascript version",
-  "main": "src/music21_modules.js",
+  "main": "releases/main.js",
   "files": [
     ".editorconfig",
     ".eslintignore",
@@ -10,6 +10,7 @@
     ".gitignore",
     ".jshintrc",
     "Gruntfile.js",
+    "tsconfig.json",
     "/css",
     "/ext",
     "/releases",

--- a/scripts/postinstall.bash
+++ b/scripts/postinstall.bash
@@ -8,8 +8,8 @@ curl https://codeload.github.com/cuthbertLab/midi-js-soundfonts/zip/master > sou
 (cd "${BASEDIR}"/soundfonts || exit 1; unzip -q soundfonts.zip && rm soundfonts.zip)
 
 # do not use the double-bracket form -- Ubuntu 16 cannot handle it and is still supported.
-if [ ! -f "${BASEDIR}/src/music21.js" ]; then
-    echo 'Symlinking music21.js for use with require'
+# if [ ! -f "${BASEDIR}/src/music21.js" ]; then
+#     echo 'Symlinking music21.js for use with require'
 
-    (cd "${BASEDIR}"/src || exit 1; ln -s ../releases/music21.debug.js music21.js)
-fi
+#     (cd "${BASEDIR}"/src || exit 1; ln -s ../releases/music21.debug.js music21.js)
+# fi

--- a/src/note.ts
+++ b/src/note.ts
@@ -819,5 +819,3 @@ export class Rest extends GeneralNote {
         return vfn;
     }
 }
-
-/* ------ TODO(msc): SpacerRest  or remove from music21p ------ */

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -6,6 +6,9 @@
         "experimentalDecorators": true,
         "moduleResolution": "node",
         "sourceMap": true,
+        "declaration": true,
+        "declarationMap": true,
+        "declarationDir": "build",
         "target": "es6"
     },
     "include": [

